File: src/services/storage_service.py
import os
import shutil
from fastapi import UploadFile
from typing import Optional
from loguru import logger
import uuid

class StorageService:
    """
    Gerenciador de Arquivos e Mídias (Imagens, Áudios, PDFs).
    Substitui o 'BlobStorageService' do .NET que usava Azure/S3.
    """
    UPLOAD_DIR = "uploads"
    
    @staticmethod
    def ensure_upload_dir():
        if not os.path.exists(StorageService.UPLOAD_DIR):
            os.makedirs(StorageService.UPLOAD_DIR)
            logger.info(f"📂 Diretório de uploads criado: {StorageService.UPLOAD_DIR}")

    @staticmethod
    async def save_upload(file: UploadFile, tenant_id: str) -> str:
        """Salva um arquivo enviado pelo UI (Agente) para o bot."""
        StorageService.ensure_upload_dir()
        
        # Estrutura por Tenant para isolamento físico
        tenant_path = os.path.join(StorageService.UPLOAD_DIR, tenant_id)
        if not os.path.exists(tenant_path):
            os.makedirs(tenant_path)
            
        filename = f"{uuid.uuid4()}_{file.filename}"
        file_path = os.path.join(tenant_path, filename)
        
        with open(file_path, "wb") as buffer:
            shutil.copyfileobj(file.file, buffer)
            
        return file_path

    @staticmethod
    def get_public_url(file_path: str) -> str:
        """Retorna a URL pública para acesso externo (Bot/Venom)."""
        # Em dev: http://localhost:8000/uploads/tenant_1/file.jpg
        clean_path = file_path.replace('\\', '/')
        return f"/{clean_path}"

File: node-version/src/services/storageService.js
const fs = require('fs');
const path = require('path');
const { v4: uuidv4 } = require('uuid');
const logger = require('../utils/logger');

class StorageService {
  static UPLOAD_DIR = path.join(__dirname, '..', '..', 'uploads');

  static ensureUploadDir() {
    if (!fs.existsSync(this.UPLOAD_DIR)) {
      fs.mkdirSync(this.UPLOAD_DIR, { recursive: true });
      logger.info(`📂 Diretório de uploads criado: ${this.UPLOAD_DIR}`);
    }
  }

  static async saveUpload(fileBuffer, originalName, tenantId) {
    this.ensureUploadDir();

    const tenantPath = path.join(this.UPLOAD_DIR, tenantId);
    if (!fs.existsSync(tenantPath)) {
      fs.mkdirSync(tenantPath, { recursive: true });
    }

    const filename = `${uuidv4()}_${originalName}`;
    const filePath = path.join(tenantPath, filename);

    fs.writeFileSync(filePath, fileBuffer);
    return filePath;
  }

  static getPublicUrl(filePath) {
    // Normaliza para URLs web
    const relativePath = path.relative(path.join(__dirname, '..', '..'), filePath);
    const cleanPath = relativePath.split(path.sep).join('/');
    return `/${cleanPath}`;
  }
}

module.exports = StorageService;
